_pages/dev/tms.md

```diff
@@ -0,0 +1,243 @@
+- `events?: AbstractType<CxEvent>[]` - List all the event classes which are to be collected and pushed to the data layer.
+- `collector?: Type<TmsCollector>` - The custom collector service implementation.
+
+To use start collecting events, you need to import the `BaseTmsModule.forRoot()`, and provide a configuration.
+Optionally, you can import the `AepModule` or `GtmModule` to leverage Spartacus' default configuration.
+
+The following is how a full configuration might look like:
+
+```typescript
+import { NgModule } from '@angular/core';
+import {
+  CartAddEntrySuccessEvent,
+  CartRemoveEntrySuccessEvent,
+  provideConfig,
+} from '@spartacus/core';
+import { NavigationEvent } from '@spartacus/storefront';
+import { AepModule } from '@spartacus/tracking/tms/aep';
+import { BaseTmsModule, TmsConfig } from '@spartacus/tracking/tms/core';
+import { GtmModule } from '@spartacus/tracking/tms/gtm';
+
+@NgModule({
+  imports: [
+    ...,
+    BaseTmsModule.forRoot(),
+    GtmModule,
+    AepModule,
+    ...
+  ],
+  providers: [
+    ...,
+    provideConfig({
+      tagManager: {
+        gtm: {
+          events: [NavigationEvent, CartAddEntrySuccessEvent],
+        },
+        aep: {
+          events: [NavigationEvent, CartRemoveEntrySuccessEvent],
+        },
+      },
+    } as TmsConfig),
+  ],
+})
+export class AppModule {}
+```
+
+In the example above, we are just providing which events should be collected by each of the configured TMS solutions.
+
+### Customizations
+
+In the example above, we are leveraging Spartacus' default configuration for both GTM and AELP TMS solutions, and providing only the minimal custom configuration.
+There are various customizations available in order to tweak how Spartacus interacts with the events and the data layer:
+
+#### Data layer
+
+Spartacus uses the standard data layer object names and types for each of the supported TMS:
+
+- [GTM](https://developers.google.com/tag-manager/devguide#datalayer) - `dataLayer: any[]`
+- [AELP](https://experienceleague.adobe.com/docs/analytics/implementation/prepare/data-layer.html?lang=en#setting-data-layer-values) - `digitalData: {[eventName: string]: any}`
+
+If you use a custom data layer object (with the standard data structure), you can configure it using the `dataLayerProperty` configuration property.
+
+If you are using a different data layer structure than the standard, you can write your own collector like this:
+
+```typescript
+import { Injectable } from "@angular/core";
+import { CxEvent } from "@spartacus/core";
+import {
+  TmsCollector,
+  TmsCollectorConfig,
+  WindowObject,
+} from "@spartacus/tracking/tms/core";
+
+@Injectable({ providedIn: "root" })
+export class MyCollectorService implements TmsCollector {
+  init(config: TmsCollectorConfig, windowObject: WindowObject): void {
+    windowObject.myDataLayer = windowObject.myDataLayer ?? [];
```

Comment:
What about changing the variable name to `window: WindowObject` , so the example will be easier for eyes and in line with official adobe/gtbb doccuemntation: 
```
window.myDataLayer = window.myDataLayer ?? [];
```

Reply:
Done. 👀